%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
%\RequirePackage[authoryear]{natbib}% uncomment this for author-year bibliography
%\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
%\usepackage{siunitx}
\usepackage[colorlinks]{hyperref} % To color blocks of text 
%\usepackage[dvipsnames]{xcolor} % More color options (see Wikibooks)  
\usepackage{changepage}
\usepackage{booktabs} % For prettier tables
\usepackage{float} % To control positions of e.g. tables or figures
%\usepackage[section]{placeins}
\usepackage{graphicx}
\graphicspath{ {../} }

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\def\includegraphic{}
%\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Dual host-parasite transcriptomes of apicomplexan Eimeria falciformis and its natural mouse host}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
%	addressref={aff1},
%	noteref={n1},
]{\inits{TK}\fnm{Totta} \snm{Kasemo}}
\author[
%	addressref={aff1},
]{\inits{SS}\fnm{Simone} \snm{Spork}}
\author[
%	addressref={aff2},
%	noteref={n1},
]{\inits{CD}\fnm{Christoph} \snm{Dieterich}}
\author[
%	addressref={aff1},
]{\inits{RL}\fnm{Richard} \snm{Lucius}}
\author[
%	addressref={aff1, aff3},                   % id's of addresses, e.g. {aff1,aff2}
	corref={aff1, aff3},                       % id of corresponding address, if any
%	noteref={n1},                        % id's of article notes, if any
	email={emanuel.heitlinger@hu-berlin.de}   % email address
]{\inits{EH}\fnm{Emanuel} \snm{Heitlinger}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
	\orgname{Department of Biology, Humboldt-Universitat zu Berlin}, % university, etc
	\street{Philippstr. 13, Haus 14},                     %
	\postcode{10115}                                % post or zip code
	\city{Berlin},                              % city
	\cny{Germany}                                    % country
}
\address[id=aff2]{%  
	\orgname{Max Planck Institute for Biology of Ageing},
	\street{Joseph-Stelzmann-Strasse 9B},
	\postcode{50931}
	\city{Cologne},
	\cny{Germany}
}
\address[id=aff3]{%  
	\orgname{Department Reproduction Biology and Evolutionary Ecology, Leibniz Institute of Zoo and Wildlife Research},
	\street{Alfred-Kowalke-Str. 17},
	\postcode{10315}
	\city{Berlin},
	\cny{Germany}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
%\note[id=n1]{Equal contributor} % note, connected to author
%\note[id=n2]
\end{artnotes}

%\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{First part title} %if any
Text for this section.

\parttitle{Second part title} %if any
Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{Parasite, apicomplexa, RNA-seq, transcriptome, life-cycle, interaction}
%\kwd{article}
%\kwd{author}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
\end{fmbox}% uncomment this for twcolumn layout
\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	INTRODUCTION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Introduction}
Text and results for this section, as per the individual journal's instructions for authors. 
%\cite{koon,oreg,khar,zvai,xjon,schn,pond,smith,marg,hunn,advi,koha,mouse}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	RESULTS
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \section*{Results}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% 	OVERVIEW READS PER SAMPLE, LAYOT AS EXPERIMENTAL OVERVIEW
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

  \section*{Experimental overview}
We performed mRNA sequencing of caecum tissue from mice infected with
the apicomplexan parasite \textit{E. falciformis}. For each replicate
sample we enriched caecum tissue from three infected mice for
epithelial cells. We extracted mRNA and constructed sequencing
libraries for different timepoints after infection and mouse strains
with assumed different immunocompetence.
%% we have to bring in Simones data first with two sentences and a
%% figure or write it like this
Additonally we obtained data for challenge infections of mice
immunized one month prior to the experiment. For the parasite oocysts
and sporozoites were included as ``environmental'' lifecycle stages.

The number of sequencing reads obtained for individual replicates
ranged from 21,517,055 (sample NMRI\_sporozoites\_rep2) to 593,396,312
(NMRI\_2ndInf\_5dpi\_rep1).

As samples and idividual replicates were sequenced in batches to
different depth and using different instrumentation (Table 1) we
performed quality controls confirming the abscence of batch effects
influencing analysis and quality of our results (additional file xyc).

\section*{Parasite and host transcriptomes can be assessed in parallel}

A maximum of 86\% (sample NMRI\_1stInf\_7dpi\_rep1) and a minimum of
0.0326\% (sample NMRI\_2ndInf\_3dpi\_rep2) of mapped reads could be
assigened to the \textit{E. falciformis} genome in samples considered
infected.

We excluded samples NMRI\_2nd\_3dpi\_rep1 and NMRI\_2nd\_5dpi\_rep2
due to an exptremely low parasite contribution to the overall
transcriptome. Technically this exclusion made it possible to obtain
read counts in agreement with a negative bionmial distributions (see
adddtional file x). Judging from their mRNA output those samples can
be considered non-infected, as both samples occured in a challenge
infection we assume that the infection was cleard or reduced to a
non-detectable output in those animmals.

On the other end of the spectrum mice not previously immunized
featured an mRNA ``holo-transcriptome'' at 7dpi with the majority of
mRNA originating from the parasite as opposed to the host (Figure 1).

%%%%%
figure 1 label
cotribution of mouse vs. parasite transcripts to the overall
transcriptional output of the analysed tissue for all samples.
%%%%

The total number of sequencing reads obtained as well as those reads
mapped to the transcriptome of either \textit{E. falciformis} and the
mouse are indicated in Table 1 for all replicates.

%% Before was not pretty. All those NA's. If people are like me they
%% hate NAs in tables ;-). I think you sould make the above table from
%% data with xtable.

%% best use a straight 2d table with... 

% label & mouse strain & parasite dpi & naive/challenge & number of
% reads & #mapped to mouse & #mapped to \textit{E. falciformis}

%% column name "naive/challenge" can change

%% as technical as it seems people will want to know what our labels
%% mean (they appear in figures throughout) therfore it is good to
%% have them in table 1

\section*{The mouse transcriptome changes in the course of infection infection}

Statistical testing for differences between infected and uninfected
mice, revealed changes in mRNA abundance becoming more pronounced
(involving more genes) at later timepoints of infection (Table 1).

The observed fold-changes were in agreement with previously published
microarry data. Fold-Change data obtained from mice at 6dpi on Agilent
microarrays (Schmidt et al 2012) or 7dpi (our RNAseq data) shows a
strong correlateion (Spearman's \rho = 333; $R^2$ from a linear model
= 444 ; see also figure 2). Considering both and technical differences
between the tow methods and the biological changes along the lifecycle
this confirms the adequacy of our methods for assessing the host
transcriptome.

We selected all mouse genes differently expressed (FDR<1\%) between
uinfected and any of the infected samples.


%%% This is from methods but all the results should be presented here:

%% Selecting at most at most 500 genes differentially abundant with
%% lowest FDR (<0.05) are selected. In the next step, the 500 mRNAs from
%% each comparison (or less) are joined. For \textit{E. falciformis} time
%% p.i. comparison, this resulted in 1618 unique genes selected as
%% differently expressed . The 22 genes in the NMRI vs C57BL/6 comparison
%% are not included in the \textit{E. falciformis} life cycle
%% analysis. The same comparisons for mouse yielded 8052 genes in total
%% and 1313 unique ones. In heatmaps, all samples, i.e. also samples
%% which themselves did not have any significantly different mRNAs
%% according to our selection, were included in hierarchical
%% clustering.



Those changes are mainly related to the onset of an
adaptive immune response.




\section*{The lifecycle of Eimeria is characterized by large changes in mRNA abundance}

%%% This is from methods but all the results should be presented here: 

%% Selecting at most at most 500 genes differentially abundant with
%% lowest FDR (<0.05) are selected. In the next step, the 500 mRNAs from
%% each comparison (or less) are joined. For \textit{E. falciformis} time
%% p.i. comparison, this resulted in 1618 unique genes selected as
%% differently expressed . The 22 genes in the NMRI vs C57BL/6 comparison
%% are not included in the \textit{E. falciformis} life cycle
%% analysis. The same comparisons for mouse yielded 8052 genes in total
%% and 1313 unique ones. In heatmaps, all samples, i.e. also samples
%% which themselves did not have any significantly different mRNAs
%% according to our selection, were included in hierarchical
%% clustering.


%% This might be figure label
%% Scale bar in heatmaps show 0 as mean mRNA abundance for
%% each gene (row). Up (green) and down-regulation (brown) denote number
%% of standard deviations from 0, i.e., row mean. 



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% 	NUMBER OF GENES DIFFERENT IN DIFFERENT COMPARISONS	
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% I like this table. Looks good and transprorts the right kind of
%% information

\clearpage
\section{mRNA abundance differences between different experimental groups}
\setlength{\tabcolsep}{8pt}
\begin{table}[H]
\small
\begin{center}
\caption{mRNA abundance differences between different experimental groups.}
%\label{tab:table3}
\begin{tabular}{*3l}    \toprule
\textit{Day post infection} & \textit{Ef} genes different & Mouse genes different \\ 
	\textit{comparisons} 	    & (FDR$\leq$1\%) &  (FDR$\leq$1\%/5\%) \\ \midrule
	NMRI 0 vs NMRI 3		& NA   & 274 \\
	NMRI 0 vs NMRI 5		& NA   & 1736 \\
	NMRI 0 vs NMRI 7		& NA   & 2802 \\
	NMRI 3 vs NMRI 5     		& 111  & 1 \\
	NMRI 3 vs NMRI 7  		& 1385 & 1407 \\ 
	NMRI 5 vs NMRI 7  		& 1895 & 873 \\ 
	C57BL/6 0 vs C57BL/6 5		& NA	& 914 \\
	Rag1-/- 0 vs Rag1-/- 5		& NA	& 45 \\ \midrule
\textit{Day post infection,} & 		 & 		 \\ 
\textit{parasite relevant comparisons} 	    & 		&  \\ \midrule
	Oocysts vs NMRI 3  	& 3310 & NA \\  
	Oocysts vs NMRI 5	& 3605 & NA \\ 
	Oocysts vs NMRI 7	& 3085 & NA \\ 
	Oocysts vs sporozoites  & 3421 & NA \\
	Sporozoites vs NMRI 3 	& 1663 & NA \\
	Sporozoites vs NMRI 5 	& 1605 & NA \\
	Sporozoites vs NMRI 7 	& 2473 & NA \\ \midrule
\textit{First and second infection} & 		 & 	 \\ 
\textit{comparisons} 	    & 		& 	\\ \midrule
	NMRI 3 1st vs NMRI 3 2nd  	& 0  & 5 \\
	NMRI 5 1st vs NMRI 5 2nd  	& 0  & 1 \\
	NMRI 7 1st vs NMRI 7 2nd  	& 0  & 902 \\
	C57BL/6 1st vs C57BL/6 2nd (day 5) & 0 &  mouse \\
	Rag1-/- 1st vs Rag1-/- 2nd (day 5) & 0 & mouse  \\ \midrule
\textit{Mouse strain comparisons} & 		 & 	 \\ \midrule
	NMRI vs C57BL/6 		& 22 	& NA* \\		
	NMRI vs Rag1-/-     		& 0 & NA* 	\\
	C57BL/6 vs Rag1-/-  		& 0   & 356 	\\ \midrule
\bottomrule
	\hline
\end{tabular}
\end{center}
\end{table}

Table 2 lists pairwise sample comparisons and number of genes with
differently abundant mRNAs per comparison."NMRI" followed by number
indicates day post infection (e.g. NMRI 3 = NMRI mouse on day 3
p.i.). Genes with Benjamini-Hochberg corrected p-values $\leq$0.01 as
implemented in edgeR are included. NAs are missing samples or not
applicable for the species. NA* is due to missing NMRI day 0 sample
from first infection. For \textit{E. falciformis} no mRNAs are
significantly differently abundant between first and second infection,
whereas in mouse there are some differences, especially on day 7
p.i. However, upon hierarchical clustering of these genes (union of
500 with lowest FDR), samples do not cluster according to first/second
infection (Figure SX ?). In comparisons between mouse strains, 22
genes are detected to be different in the parasite data. The
difference is however not between Rag1-/- mice as expected, but
between NMRI and C57BL/6. For mouse....

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\clearpage
\subsection{Life cycle analysis... \textit{E. falciformis}}
\begin{figure}[h!]
%\begin{center}
%\centering
	\includegraphics[width=\linewidth]{EfLifecycleHeatmap.pdf}  
	\caption{\textit{E. falciformis} mRNAs with significantly different abundance at different times p.i. in NMRI mice (see Table 2). Samples cluster according to expected developmental state of the parasite, without distinguishing between first and challenge infection. Hierarchical clustering was set to apply seven gene clusters.}
%\end{center}
\end{figure}

We applied hierarchical clustering on \textit{E. falciformis} mRNAs which were differently abundant at different timepoints p.i., including comparisons with oocysts and sporozoites (union of 500 with lowest FDR). Samples from day 7 p.i. form a separate cluster (NMRI mice only), not distinguishing between first and challenge infection. For these samples, the pattern is most pronounced in gene cluster 1 (up), and 4 and 6 (down). Within cluster 1, the second replicate from the challenge infection has a deviant profile (compare mouse data in Figure 2). Distinct groups of genes also define sporozoites (cluster 5 up) and oocysts (cluster 3 up, cluster 6 down). mRNA profiles on days 3 and 5 p.i. from all three mouse strains cluster together. These samples are distinct from oocysts, NMRI day 7 p.i., and sporozoites. Sporozoites cluster closer with days 3 and 5 than with oocysts or day 7.


\clearpage
\subsection{Parasite development reflected in mouse data}
\begin{figure}[h!]
%\begin{center}
%\centering
\includegraphics[width=\linewidth]{MmLifecycleHeatmap.pdf}  
\caption{Mouse mRNAs with significantly different abundance at different times p.i. in NMRI mice (see Table 2). Samples cluster according to expected developmental state of the parasite, without distinguishing between first and challenge infection. Rag1-/- mouse samples cluster together. Hierarchical clustering was set to apply four gene clusters.}
%\end{center}
\end{figure}

When clustering mouse mRNAs which were different in time p.i. comparisons in Table 2, three out of four samples from day 7 p.i. (NMRI only) cluster together. These samples are characterized by gene cluster 3 (up) as well as 2 and 4 (down). The fourth day 7-sample (challenge infection, second replicate) clusters with day 3 and 5 samples. Upon visual inspection of this sample in gene cluster 4, it diplays a profile similar to non-infected mice. The same sample is abnormal in the parasite profile (Figure 1). NMRI and C57BL/6 uninfected samples (0dpi) cluster together, defined by gene clusters 4 (up), and 1 (down). Rag1-/- day 0 samples are however less pronounced in gene cluster 1. Day 3 and 5 samples cluster together, with Rag1-/- forming a separate group and Rag1-/- non-infected most distant in this sample cluster.




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\clearpage
\subsection*{GO term enrichments in heatmap clusters}
The annotations referred to here are inferred from orthologs in other \textit{Eimeria} spp. or
in \textit{T. gondii}

\subsubsection*{Data overview}
In total, xx reads were sequenced. For each sample, mouse read numbers were $10^7$ - $10^8$
Quality filters applied resulted in xx reads being removed from further analysis

\subsubsection*{Preparation for invasion in oocysts}
The mRNA profile in the oocyst stage is mainly determined by highly abundant genes in cluster 4.
Overrepresented GO-terms in this cluster are enriched by ortholog genes to 
peptidases, microneme localized proteins reported to be involved in invasion, 
genes associated with adhesion in protozoans (and with clotting in higher eukaryotes),
and genes that are annotated to be involved in amino acid biosynthesis.
  Aminopeptidase N ('related' annotation) is the reported ortholog for three genes with 
abundant mRNAs in oocysts. In humans, this enzyme has been reported to cleave peptides bound to 
major histocompatibility complex, MHC, II (UniProt reference if we want to keep this... but does any
secretion happen from oocysts...? Or is this too far-fetched to be interesting?).

A Thrombospondin type 1 domain-containing protein ortholog is highly abundant in cluster 4 (high 
abundance in oocysts). Thrombospondin type 1 domains have been reported in \textit{E. tenella} microneme
localizing proteins, MIC, e.g. MIC4 (Tomley01) In \textit{E. tenella} MIC4 mRNA was reported in 
sporozoites where it localizes to the apical end, and in late schizonts and late oocyst stages,
when sporozoites are forming. (Tomley01). For the same gene, the \textit{T. gondii} annotation is 
Sushi domain-containing protein, which is also the ortholog annotation of another gene in this 
cluster. In the related malaria parasite \textit{P. falciparum} the apical sushi 
protein, ASP, (which has a sushi domain) localizes to micronemes in merozoites but not 
other stages (OKeeffe05). 
Limulus clotting factor C, Coch-5b2 (Cochlin) and Lgl1, LCCL, 
(syn. F5/8 domain) domains are associated discoidin lectin domains and thereby with adhesion.
(Pfam entries for 'LCCL domain' and 'Discoidin domain', May 2016). 
Taken together, this indicates that the thrombospondin and sushi-domain genes 
(EfaB\_MINUS\_4114.g412 and EfaB\_PLUS\_1425.g183) are involved in sporozoite invasion in 
\textit{E. falciformis} and that the mRNAs are transcribed and available before excystation. 
A role in merozoite re-invasion in \textit{E. falciformis} is not indicated by our data.
The LCCL domain annotation and thrombospondins role in higher eukaryotes also indicates 
that adhesion or preparation for adhesion is important in oocysts. 
We suggest (speculate...?) that the thrombospondin annotated ortholog and the LCCL domain-containing protein 
(EfaB\_MINUS\_11233.g986) are involved in cell adhesion in \textit{E. falciformis}.
%One examples
%is found in the slime mold \textit{Dictyostelium discoideum)} in which the LCCL domain
%is part of the mold's discoidin adhesion protein. (Pfam entry for 'discoidin domain', May 2016)

\subsubsection*{Amino acid biosynthesis in oocysts}
High abundance of aminotransferase mRNAs indicate amino acid biosynthesis or preparation for 
the same in oocysts (cluster 4). 
We identify D-3-phosphoglycerate dehydrogenase and alanine dehydrogenase orthologs, which 
are enzymes contributing to L-serine and L-alanine production, respectively. A putative 
\textit{Eimeria} spp. cystathionine beta-synthase, CBS, in this cluster also indicates 
de novo cysteine production. Alkyl sulfatase mRNA is also abundant in oocysts. Generally, 
this enzyme enables an organism to exploit organic sulfur to produce and incorporate 
inorganic sulfur into the amino acids cysteine and methionine, when no inorganoc sulfur 
is available. 

'Embryonic development'
Nicalin 1, patched family protein (hedgehog)  


\subsubsection*{Oocysts contain mRNA for fatty acid catabolism}
MmgE/PrpD is overrepresented in oocysts. The enzyme is important for propionate catabolism 
in the 2-methylcitric acid cycle and has been shown to be used by the 
intestinal intracellular bacterium \textit{Salmonella typhimurium} to generate 
pyruvate (Horswill99). 
Propionate is one of two most abundant small-chain fatty acids 
in the gut along with butyrate. Both fatty acids are largely produced as 
degradation products from food by commensal bacteria (Sun13). Sharing the intestines as a niche
with \textit{S. typhimurium} it is possible that also \textit{E. falciformis} uses Mmg/PrpD
to exploit available propionate for pyruvate production.

\subsubsection*{Oocyst highly abundant mRNAs are downregulated in sporozoites}
Interestingly, the genes described above which are thought to be involved in amino acid 
biosynthesis and invasion are highly abundant in oocysts but are underrepresented in schizont
stages (day 3 and day 5 samples) and even in sporozoites. An average abundance was detected on 
day 7 for these genes, indicating a role in either gametes or early oocyst formation.
This pattern supports the suggestion that these specific mRNAs (cluster 4) for invasion 
and biosynthetic processes are prepared (and possibly expressed) in the oocyst stage but 
are no longer detectable in the cell at the timepont when the protein is assumed to be 
in use (sporozoites and merozoite stages). 
Therefore, correlating mRNA prevalence with biological function at the timepoint when mRNAs 
are detected must be done with care. 

\subsubsection*{Down in sporozoites and oocysts -->  cluster 3..........}
Oocysts: profile for 3, 5, 6:

Sporozoites: 3 and 5

Day 7: 5 and 7

Specific genes....
Enolase 2, encoded by Eno2, is among the downregulated genes in oocysts and sporozoites.
In \textit{T. gondii} the paralog Eno1 is strongly associated with the cyst (bradyzoite) stage
and Eno2 is associated with tachyzoite stages (kibe05). It is therefore expected that this mRNA is
underrepresented in oocysts and our data also show that the same is true in sporozoites for
\textit{E. falciformis}. (TK: If important we could look specifically for Eno1 in cluster 4).








%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%% DAY xx   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Down in sporozoites and oocysts -->  cluster 3..........}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%% DAY 7  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Motility-related mRNAs indicate gamete development on day 7}
Two clusters contain genes with mRNAs highly abundant on day 7 p.i; cluster 1 and 2.
Dynein, kinesin and tubulin are annotations highly represented among orthologs of genes 
in both these clusters. The annotations indicate an important role for motility at this
timepoint, probably reflecting development of microgametes.
In addition, in cluster 2, there are two 'EF-hand domain containing proteins' annotations as well 
as caltractin, centrin-1, and troponin annotations. Caltractin and 
centrin-1 are associated with the centrosome and structure and function of microtubuli in mammals, 
and troponin is linked to muscle function (UniProt). 
Also potentially linked to motility is the occurence of growth arrest specific protein 8,
Gas8, which in the mouse has been reported to be highly 
expressed in the testes and important for mouse sperm function (Yeh02). 

Other genes among the 38 indicate carbon fixation (glycolysis/gluconeogenesis) or 
conversions of nucleoside phosphates. In addition, a Ras family protein, RNA 
polymerase II transcription initiation factor and Sec23 and Sec24 were among orthologs 
identified in \textit{E. falciformis} cluster 2.

In cluster 1, carbon metabolism genes are represented by 6-phosphogluconate dehydrogenase 
and glycogen phosphorylase family protein 1. UDP-glucose 4-epimerase and amiloride-sensitive 
amine oxidase are reported as upregulated in gametocytes in \textit{E. tenella} by 
RNA-seq (Walker15) and suggested by those authors to play a role in cyst wall synthesis.

\subsubsection*{Microneme proteins highly expressed on day 7 p.i.}
Unintuitively for a protozoan organism, seven out of eight GO biological process terms
in cluster 1 are associated with wound healing and blood coagulation.
An explanation is offered by some of the orthologs to the three \textit{E. falciformis} genes
responsible for these terms. In protozoa, e.g., other \textit{Eimeria} spp. and 
\textit{Toxoplasma gondii} orthologs are annotated as 'Micronemal protein MIC4,
related' (E. tenella) and more generally for several other protozoa, 'PAN domain containing proteins'.
The PAN domain is found in the plasminogen/hepatocyte growth factor family and in coagulation 
factor XI family (REF), explaining why terms related to blood coagulation are enriched by these genes.
Later publications on \textit{T. gondii} (Marchant12) also associate PAN domains and proteins in 
apicomplexan parasites with micronemes and therefore invasion. In our case, this is peculiar, since 
the enrichment appears on day 7 p.i.. A possible role at this timepoint is suggested by work on 
the fungi \textit{Sclerotinia sclerotiorum} where Yu et al. reported an important role for PAN domain
proteins in cell wall integrity (Yu12). This role for MIC proteins has to our knowledge 
not been investigated in apicomplexan parasites. 
The PAN domain domain has also been reported to be common in nematodes such as 
\textit{Caenorhabditis elegans}, however the function is not understood. (Thordai99)
The other two GO terms in the cluster of day seven upregulated genes are DNA replication and DNA 
replication initiation, which most likely reflects late stage schizogony or gamete formation.
Six genes contribute to this enrichment and orthologs are either annotated as DNA replcation
licencing factors, DNA polymerases or minichromosome maintenance proteins 2/3/5/7, Mcm2/3/5/7.


\subsubsection*{Gene and sample patterns by hierarchical clustering}
Samples (columns) cluster into two major clusters where day 7 p.i. samples form one group distinct from
other samples. In the second group, oocsts and sporozoites are distinct and sporozoites cluster
most closely with day 3 and 5 p.i. samples. Day 3 and 5 p.i. samples also cluster into two groups,
of which one contains all NMRI day 5 p.i. samples. Apart from this, the two day 3 and 5 p.i. sample
clusters have no obvious patters. 

For gene clusters (rows), the two groups with high mRNA abundance on day 7 p.i. (cluster 1 and 2)
do not cluster most closely with each other, but with the cluster for high mRNA abundance in oocysts
(cluster 1 association) and with the cluster for high mRNA abundance in sporozoites (cluster 2 
association).


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	DISCUSSION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Discussion}
In our analysis we demonstrate which biological processes are dominant in different
life cycle stages of \textit{E. falciformis} in the mouse. The RNAseq transcriptome provided here 
allows for detailed analysis of genes involved in those processes, providing candidates for 
life stage specific marker in \textit{Eimeria} spp. research. 

\ldots
\ldots
\ldots


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	METHODS
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Methods}
\subsection{Mice and infection procedure}
Three strains of mice were used in our experiments: NMRI (Charles
River Laboratories, Sulzfeld, Germany), C57BL/6 (), and Rag1-/- on
C57BL/6 background (gift from Susanne Hartmann, FU?).  Animal
procedures were performed according to the German Animal Protection
Laws as directed and approved by the overseeing authority Landesamt
fuer Gesundheit und Soziales (Berlin, Germany). Animals where infected
as described by Schmid et al., (Schmid12), but tapwater was used
instead of PBS for administration of oocysts. Briefly, NMRI mice were
infected two times, which will be referred to as first and second
infection. For the first infection, 150 sporulated oocysts were
administered in 100 $μ$L by oral gavage. During the first infection of
60 mice, all animals were weighed every day. On day zero, before
infection, as well as on day three, five and seven days post
infection, dpi, caeca from 3-4 sacrificed mice per time point were
collected. Epithelial cells were isolated as described in Schmid et
al.(schmid12). For challenge infection, mice recovered for four weeks
before second infection.  Recovery was monitored by weighing and
visual inspection of fur. For the second infection, 1500 sporulated
oocysts were applied by oral gavage. Three mice were used as
non-second infection control, referred to as day 0, second infection.

\subsection{Oocyst purification for infection and sequencing}
Sporulated oocysts were purified by flotation from feces stored in
potassium dichromate and administered orally in 100 uL tapwater. One
\textit{E. falciformis} isolate, \textit{E. falciformis} Bayer
Haberkorn 1970, was used for all infections and parasite samples. The
strain is maintained through passage in NMRI mice in our facilities as
described elsewhere (schmid12).

\subsection{Sporozoite isolation}
Sporozoites were isolated from sporocysts by excystment. For this,
sporocysts were incubated at 37$°$C in DMEM containing 0.04\%
tauroglycocholate (MP Biomedicals) and 0.25\% trypsin (Applichem) for
30 min. Sporozoites were purified by the method of Schmatz et al
(schmatz--).

\subsection{RNA extraction}
Total RNA was isolated from infected epithelial cells, sporozoites and
sporulated oocysts using Trizol according to the manufacturer’s
protocol (Invitrogen). High quality \emph{what is the meaning of 'high
  quality' here?}  RNA was used to produce an mRNA library using the
Illumina’s TruSeq RNA Sample Preparation guide.  \emph{stolen from
  genome paper} Sporozoites were stored in 1 mL Trizol until
RNA-isolation.  Total RNA was isolated using the PureLink RNA Mini Kit
(Invitrogen) and reverse transcribed into cDNA.

\subsection{Sequencing, sequence quality assessment and alignment}
cDNA samples were sequenced by either GAIIX or Illumina Hiseq 2000 as
specified in SI xx (both unstranded). A fastq\_quality\_filter
(FASTQ-toolkit, version 0.0.14, available at
https://github.com/agordon/fastx\_toolkit.git) was applied to Illumina
Hiseq 2000 samples after replacing "N" bases by "." annotation.  A
phred score of 10 was applied. We further set q = 60. These settings
require that nine out of ten bases or more are correct in at least
60\% of the bases for each read.

\subsection{Alignment and reference genomes}
We used the published \textit{Mus musculus} mm10 assembly (Genome
Reference Consortium Mouse Build 38, GCA\_000001635.2) as reference
genome including annotations for mouse data. The
\textit{E. falciformis} genome (Heitlinger14) was downloaded from
ToxoDB (Gajria07). For the alignment, the mouse and parasite genome
files were merged into a dual reference genome, and files including
mRNA sequences from both species were aligned against the dual
reference genome using TopHat2 (version 2.0.14, Trapnell09) with -G
specified, and a Bowtie2 (version 1.1.2, Langmead12) index of the dual
genome. Single-end and pair-end sequence samples were aligned
separately with library type 'fr-unstranded' specified for pair-end
samples. Import into R was enabled by the R package Ballgown, which
requires bam files to be processed by Tablemaker (Frazee15), in our
case used with -qW -G specified. Tablemaker in turn makes use of
Cufflinks (version 2.1.1, Trapnell10).

\subsection{Differential mRNA abundance, data normalisation and sample exclusions}
Count data was normalized using the R-package edgeR (version 3.14.0;
cite) with the upperquartile normalisation method. Briefly, genes with
zero coverage in all samples (libraries) are removed and normalisation
factors are calculated for the 75\% quantile for each library. This
normalisation is suitable for read densities following a negative
binomial distribution. Two samples contradicted this assumption
(parasite data) for later modelling and both mouse and parasite data
from these samples were excluded from further analysis:
NMRI\_1st\_3dpi\_rep1 and NMRI\_2nd\_5dpi\_rep1 (SI ...). The method
then fits a generalized linear model (GLM with a negative binomial
link function) for each gene (glmFit) and then performs likelihood
ratio tests for models w or w/o focal factor (glmLRT).

%% ........... include outcome of MDS analysis here?
%% No, results never are presented in methods.

\subsection{Selection of differentially abundant mRNAs nad hierarchical clustering}
A selection of differently abundant mRNAs are used for hierarchical
clustering of \textit{E. falciformis} life cycle relevant genes. In
each comparison (see Table 3), the union of the at most 500 genes
differentially abundant with lowest FDR (<0.05) are selected. In the
next step, the 500 mRNAs from each comparison (or less) are
joined. For \textit{E. falciformis} time p.i. comparison, this
resulted in 1618 unique genes selected as differently expressed . The
22 genes in the NMRI vs C57BL/6 comparison are not included in the
\textit{E. falciformis} life cycle analysis. The same comparisons for
mouse yielded 8052 genes in total and 1313 unique ones. In heatmaps,
all samples, i.e. also samples which themselves did not have any
significantly different mRNAs according to our selection, were
included in hierarchical clustering. Scale bar in heatmaps show 0 as
mean mRNA abundance for each gene (row). Up (green) and
down-regulation (brown) denote number of standard deviations from 0,
i.e., row mean. Hierarchical clustering was performed using with
Euclidean distances, using complete linkage ('complete', R package
base).

%% the paragraph above has results. All those number 



\subsection{Differentially abundant mRNAs used for hierarchical clustering}
All analyses were performed in R (cite R-core). Complete scripts are
available at git@github.com:derele/Ef\_RNAseq.git tagged as version
1.0. 
%% We will make this available as a git webpage probably

\setlength{\tabcolsep}{10pt}
\begin{table}[H]
\small
\begin{center}
\caption{Genes used for hierarchical clustering of mRNAs differently abundant depending on time p.i..}
\begin{tabular}{*3l}    \toprule
	\textit{Data description} & \emph{E. falciformis} genes & Mouse genes	\\ \midrule
	Sum of 1st infection NMRI sample differences	& 4935 & 8052 \\ 
	(including oocysts and sporozoites if appl.)	\\	
	Used in hierarchical clustering (heatmap)  	& 1618 & 1313 \\ 	\bottomrule	
\hline
\end{tabular}
\end{center}
%\end{adjustwidth}
\end{table}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
    Text for this section \ldots

\section*{Acknowledgements}
  Text for this section \ldots
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )
% for author-year bibliography (bmc-mathphys or spbasic)
% a) write to bib file (bmc-mathphys only)
% @settings{label, options="nameyear"}
% b) uncomment next line
%\nocite{label}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

%\begin{figure}
 %\begin{picture}
 % \includegraphics[width=0.5\linewidth]{ef-heatmap}
 %\end{picture}
%\end{figure}

\section*{Figures}
\begin{figure}[h!]
%  \includegraphics[width=0.5\linewidth]{efmm-ratio}
  \caption{\csentence{Parasite read sequences per mouse read sequences for each experimental condition.}
  The more advanced the infection is, the higher the ratio of parasite reads is, reflecting parasite replication between day zero and day seven. ......compare 1st and 2nd ....... patterns between Rag and NMRI/C57BL/6?  ........ Values are mean of replicates (n=2, if * n=1) on log10 scale. Each sample (replicate) consists of mRNA from three different mice.}
  \end{figure}

 % \begin{figure}[h!]
 % \includegraphics[width=0.7\linewidth]{ef-heatmap}
 % \caption{\csentence{Parasite genes with different mRNA abundance between samples, sorted into five clusters by hierarchical clustering (method = complete, distance = euclidean).}
 %     \textit{E. falciformis} samples from seven days p.i. cluster together (NMRI mice only). These samples have distinct mRNA abundance patterns in all gene clusters, although more pronounced in clusters 1 (up) and 4 (down). Distinct groups of genes also define sporozoites (cluster 5, up) and oocysts (cluster 2, up). mRNA profiles on days three and five p.i. from all three mouse strains cluster together. These samples are distinct from oocysts, NMRI day 7 p.i., and sporozoites, however closest to the latter. On scale bar, 0 is mean mRNA abundance for each gene. Up and downregulation is standard deviations from mean.}
 %     \end{figure}

%\begin{figure}[h!]
%  \includegraphics[width=0.7\linewidth]{mm-heatmap}
%  \caption{\csentence{Mouse genes with different mRNA abundance between samples, sorted into four clusters by hierarchical clustering (method = complete, distance = euclidean).}
%Three out of four samples from day 7 p.i. (NMRI only) cluster together. These samples are characterized by genes in cluster 2 (up), 3, and 4 (down). NMRI and C57BL/6 uninfected samples cluster together (left), defined by clusters 3, 4 (up), and 1 (down). Rag1-/- samples cluster together. These samples share a weak downregulation of most genes in cluster 2 as well as upregulation of a small group genes in the same cluster. Uninfected Rag1-/- samples are separated from infected ones with distinct profiles in clusters 1, 3, and 4. On scale bar, 0 is mean mRNA abundance for each gene. Up and downregulation is standard deviations from mean.}
%      \end{figure}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\begin{table}[h!]
\caption{Sample table title. This is where the description of the table should go.}
      \begin{tabular}{cccc}
        \hline
           & B1  &B2   & B3\\ \hline
        A1 & 0.1 & 0.2 & 0.3\\
        A2 & ... & ..  & .\\
        A3 & ..  & .   & .\\ \hline
      \end{tabular}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Raw and normalized counts }
  Raw and normalized counts for individual samples in our
  study (two compressed csv files).
  
  \subsection*{Additional file 2 --- Results of statistical tests (edgeR) }
  Fold-changes, p-values and false discovery rates (adjusted p-values)
  are given for all tested contrasts (one compressed csv file). 
  
  \subsection*{Additional file 3 --- Additional methods and resuts }
  % this would be cool but requires special referencing in the text
  % (hard to keep track and hard to fit in the journals referencing
  % system). But we can try.
  Document containing additonal figures and summary tables (pdf).

\subsection*{Additional file 4 --- Results of enrichment analyses (topGO)}
  Tables listing all tested gene sets and resulting significatn GO
  terms.
  
  
\end{backmatter}
\end{document}
