%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
\usepackage{amsthm,amsmath}
\RequirePackage{natbib}
%\RequirePackage[authoryear]{natbib}% uncomment this for author-year bibliography
\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
%\usepackage{siunitx}
\usepackage[colorlinks]{hyperref} % To color blocks of text 
%\usepackage[dvipsnames]{xcolor} % More color options (see Wikibooks)  
\usepackage{changepage}
\usepackage{booktabs} % For prettier tables
\usepackage{float} % To control positions of e.g. tables or figures
%\usepackage[section]{placeins}
\usepackage{graphicx}
\usepackage[flushleft]{threeparttable}

% to color table rows
\usepackage{color, colortbl}
\graphicspath{ {../} }

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\def\includegraphic{}
%\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Dual host-parasite transcriptomes of apicomplexan Eimeria
  falciformis and its natural mouse host}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[ addressref={aff1},] {\inits{TK}\fnm{Totta} \snm{Kasemo}}
\author[ addressref={aff1},] {\inits{SS}\fnm{Simone} \snm{Spork}}
\author[ addressref={aff2},] {\inits{CD}\fnm{Christoph} \snm{Dieterich}}
\author[ addressref={aff1}, ]{\inits{RL}\fnm{Richard} \snm{Lucius}}
\author[ addressref={aff1, aff3},              
  corref={aff1},                   
  email={emanuel.heitlinger@hu-berlin.de}]{\inits{EH}\fnm{Emanuel} \snm{Heitlinger}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{
	\orgname{Institute of Biology, Humboldt-Universitat zu Berlin}, 
	\street{Philippstr. 13, Haus 14},
	\postcode{10115}                 
	\city{Berlin},                   
	\cny{Germany}                    
}
\address[id=aff2]{  
	\orgname{Max Planck Institute for Biology of Ageing},
	\street{Joseph-Stelzmann-Strasse 9B},
	\postcode{50931}
	\city{Cologne},
	\cny{Germany}
}
\address[id=aff3]{  
	\orgname{Leibniz Institute of Zoo and Wildlife Research},
	\street{Alfred-Kowalke-Str. 17},
	\postcode{10315}
	\city{Berlin},
	\cny{Germany}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
%\note[id=n1]{Equal contributor} % note, connected to author
%\note[id=n2]
\end{artnotes}

%\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
%\parttitle{First part title} %if any
Apicomplexan parasites such as \textit{Plasmodium} spp., \textit{Toxoplasma gondii} and \textit{Eimeria} spp. cause disease in humans, livestock and wild animals. The genus \textit{Eimeria} contains thousands of niche specific intracellular parasites, including several species which cause losses in poultry industries. \textit{Eimeria falciformis} naturally infects the cecum of mice. Infecting one of the best studied and available animal models in biological research, \textit{E. falciformis} constitutes a perfect model to investigate \textit{Eimeria} parasites. However, much is still unknown about the parasite’s basic biology and no in vitro culture has been established for the full life cycle. We have performed a dual RNA-seq transcriptome study of the full life cycle in the mouse and of in vitro cultured sporozoites and oocysts. Drastic differences are seen in both parasite and host at three time-points post infection. Comparisons between immunocompetent and immunocompromised mice show differences in oocyst output as well as transcriptional differences indicated by Gene Ontology, GO, enrichments. In mouse, TGF-beta, EGF, TNF and IL-1 and IL-6 are examples of genes reacting differently depending on mouse immune status. Parasite transcriptomes have distinct profiles early and late in infection, characterized by biosynthesis and motility, respectively. Sporozoites and oocysts can also be identified by their respective transcriptional profiles. Taken together, the analysis highlights general patterns in the parasite’s life cycle and links them to biological processes. It also lays the ground for detailed analysis of specific parasitic stages and the genes relevant in them. The use of hosts with different immune competence highlights the role of adaptive and innate immunity and offers a source for in-depth analysis of these responses.


%\parttitle{Second part title} %if any
%Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{Parasite, apicomplexa, RNA-seq, transcriptome, life-cycle, interaction}
%\kwd{article}
%\kwd{author}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
\end{fmbox}% uncomment this for twcolumn layout
\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	INTRODUCTION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Introduction}
\textit{Eimeria falciformis} is an intracellular parasite in the phylum Apicomplexa. There are several thousands of apicomplexan species, including \textit{Toxoplasma gondii}, causative agent of toxoplasmosis, \textit{Plasmodium} spp., causing malaria and \textit{Cryptosporidium} spp, which cause cryptosporidosis. Coccidiosis is a disease in livestock and wildlife caused by coccidian parasites such as \textit{Eimeria} spp., including several economically problematic chicken parasites, e.g., \textit{E. tenella} (cite?). A useful model for studying \textit{Eimeria} spp. is \textit{E. falciformis}, which naturally infects wild mice, \textit{Mus musculus}, and laboratory mice. 

\textit{Eimeria} spp. infect the gut and are highly niche specific within parts and cell types of the intestines (cite?). There, these monoxenous parasites go through asexual (schizogony) and sexual reproduction which results in hosts releasing high numbers of resistant oocysts. (cite?) When a mouse ingests \textit{E. falciformis} oocysts, one sporulated oocyst releases eight infective sporozoites inside the host, which can infect epithelial crypt cells of the caecum and colon. Within the epithelium, so called merozoite stages form in several rounds of schizogony. (mesfin78) Parasite numbers increase drastically during schizogony, which is not completely synchronized and the exact number of schizogony rounds is not clear. Mesfin and Bellamy (1978) detected four schizont stages, but this has later been contradicted by our (? Richard?) unpublished data and others (???). Oocysts are first detected in faeces on day six to seven after infection (cite?), and it peaks on day eight to nine p.i. Oocysts are not detectable after day 13-15 p.i. Oocysts form as merozoites differentiate into gametes, which fuse and form a zygote. Immature, unsporulated oocysts are shed into the environment where they mature into sporulated, infective oocysts.. (cite?)

Studies on host responses to this infection have shown that in \textit{E. falciformis} infection in laboratory mice, IFNy plays an important role and is upregulated. (schmid14, more...?). In IFNy deficient (or receptor deficient) mice, oocyst output is reduced but at the expense of the host, which displays larger weight losses and intestinal pathology in this model. In the IFNy defincient model, the phenotype was recovered by blocking IL-17A and IL-22 signalling, whereas only blocking IL-22 signalling recovered the oocyst shedding but had no effect on weight loss. (stange12) (should we bring up acquired immunity topic here?)

These studies demonstrate the complex relationships between parasite and host, e.g., exemplifying that larger pathology in the host is not necessarily beneficial for the parasite, which is then less successful in producing oocysts. Several studies address host parasite interactions in apicomplexan parasites (e.g. walkerxx, hehlxx, reidxx, schmid14), but often with a strong focus on either host or parasite. In Plasmodium, reidberrimanxx have used large datasets to correlate host and parasite transciptional changes and protein domain information to infer host-parasite interactions at the molecular level. To our knowledge, in \textit{Eimeria} spp., no studies with simultaneous host and parasite data have been reported. 

By applying dual RNA-seq and producing host and parasite transcriptomes from the same samples and tissue, we provide a first dataset which allows analysis of host and parasite mRNA profiles at several time-points post infection. Not only can we for the first time provide transcriptomes for the full parasite life cycle. In addition, having host transcriptome data from the same samples, we analyse biological processes in both species and discuss possible interactions of interest. This is particularly useful since the mouse is such a well annotated organism. 
Using the mouse model, we also had the possibility to include a mouse strain which lack a gene for maturation of T and B cell receptors, Recombination-activating gene 1, Rag1. Both naive immune competent and T and B cell deficient mice were infected, and both groups were challenge infected after recovering from the first infection. This design allows a full comparison of infections between immune competent mice and mice with severely limited adaptive immunity, and analysis of how these differences affect the parasite transcriptomes and reproductive success. We contextualize the parasite data by testing for enrichment of \textit{E. falciformis} specific genes, apicomplexa specific genes or widely conserved genes in gene clusters which characterize different life cycle stages. We also correlate our RNA-seq data with published transcriptomes from the chicken parasite \textit{E. tenella} and \textit{T. gondii}. 

Taken together, the present study provides the first full life cycle transcriptome of an Eimeria spp. which naturally infects the research model M. musculus. In addition to providing a description of changes in the life cycle, we present matching host transcriptomes and analyze the host biological processes which take place during infection. This data is a source for in-depth analyses and hypothesis testing of apicomplexan parasite infections in one of the best annotated model animals at hand. 
%\cite{koon,oreg,khar,zvai,xjon,schn,pond,smith,marg,hunn,advi,koha,mouse}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	RESULTS & DISCUSSION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Results and Discussion}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% 	OVERVIEW READS PER SAMPLE, LAYOT AS EXPERIMENTAL OVERVIEW
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection*{A dual transcriptomics experiment}

We performed mRNA sequencing of caecum epithelial tissue from mice infected with
the apicomplexan parasite \textit{E. falciformis}. Oocysts and
sporozoites were included as "environmental" stages and processed
in vitro. To follow the life cycle of the parasite, we
compare different time-points after infection. We additionally used different mouse strains which display different immunocompetence in infection trials, measured by oocyst output, in order to assess the influence of host immunocompetence on parasite development (Figure 1a and 1b). Immune competent NMRI mice were infected and sampled at three time-points post infection, p.i. We also infected mice lacking Rag1 (Rag1-/-) and compared them to the parental C57BL/6 mouse line. Rag1-/- mice lack mature B and T lymphocytes, which is taken as a proxy for absent adaptive immunity. All three mouse strains were used for infections of both naive mice and previously infected (and recovered) animals (onward referred to as "challenged").
Basic phenotyping (Figure 1a) showed differences in oocyst numbers between the immunocompetent (C57BL/6) and immunodeficient (Rag1-/-) host strains in naive and challenge infected mice, as well as differences in amount of parasite 18S sequences determined by quantitative reverse transcription PCR (RT qPCR). These data suggest reduced parasite numbers and reduced reproductive success in immunocompetent, challenge infected mice. In contrast, no difference was seen between first and challenge infection in immunodeficient Rag1-/- mice. Immunocompetent NMRI mice were infected with a higher dosis of sporulated oocysts (150 in first infection and 1500 in challenge infection, which was required to see a response. Data not shown), and a drastic reduction of oocysts in faeces was seen in challenged mice. Oocyst numbers in faeces peaked on days 8-9 and all mice had cleared the infection by day 14. Asexual replication of \textit{E. falciformis} intestinal stages is reflected by the percentage of parasite reads sequenced per time-point post infection (Figure 1b) and this was confirmed by RT qPCR of parasitic 18S (Figure 1c).
We thus use dual RNA-seq to analyse the life cycle of \textit{E. falciformis} under the influence of different host immune capacity at early and late stages of infection. We used an experimental design which allows to compare infections at 5 days post infection (dpi) for all experimental conditions (NMRI, C57BL/6 and Rag1-/- mouse strains in naive and challenge infection). Additional time-points 3 dpi and 7 dpi were analysed from NMRI mice (Figure 1d).

\subsection*{Parasite and host dual transcriptomes can be assessed in parallel}
We purified host and parasite RNA from infected tissues, and we here demonstrate that even early in infection there is sufficient parasite material for detection by RNA-seq. Each replicate sample was enriched for caecum epithelial tissue and pooled from three mice. mRNA was extracted and sequencing libraries prepared. Two biological replicates were used for all but two conditions (with one and three replicates, respectively). Libraries were sequenced on several lanes of Illumina GAIIX (13 samples) and HiSeq machines (14 samples) and mapped to both mouse and parasite genomes simultaneously to avoid spurious assignments of reads in ultra conserved genomic regions. As samples and individual replicates were sequenced in batches to different depth and using different instrumentation (Table 1) we performed quality controls (additional files xyz). These confirm the absence of batch effects influencing analysis and quality of results. Total numbers of sequenced reads as well as reads mapped to either the \textit{E. falciformis} genome or the mouse genome are indicated in Table 1 for all replicates. The number of total read mappings for individual replicates ranged from 25,362,739 (sample Rag\_1stInf\_0dpi\_rep1) to 139,749,046 (NMRI\_1stInf\_7dpi\_rep2). At the latest time-points, 7 dpi, the overall mRNA output of the sampled caecum tissue is dominated by parasite material with proportional parasite mRNA abundance of 77\% (in NMRI\_1stInf\_7dpi\_rep2) and 92\% (in NMRI\_1stInf\_7dpi\_rep1). (Figure 1b) suggesting heavily infected tissues and large parasite numbers at this time-point.  

\subsection*{Exclusion of samples with uncertain infection status}
Existing analysis methods of RNA-seq data (e.g. edgeR, DEseq) require certain distributions of data for reliable normalization and differential gene expression analysis. Low numbers of reads is a potential source of problematic, noisy distributions. We report a mapping to the parasite genome of maximum 92\% (sample NMRI\_1stInf\_7dpi\_rep1) and a minimum of 0.064\% (sample NMRI\_2ndInf\_3dpi\_rep2) in samples considered infected (Table 1). We excluded samples NMRI\_2nd\_3dpi\_rep1 (0.012\%) and NMRI\_2nd\_5dpi\_rep2 (0.023\%) due to low parasite contribution to the overall transcriptome. Technically, this exclusion made it possible to obtain read counts in agreement with a negative binomial distribution (see additional file x). It is also likely that the number of reads in the excluded samples would have been insufficient to fully normalise these datasets to those with the highest parasite contributions. From a biological point of view, both excluded samples are samples from challenge infection and it is likely that the infection had been cleared or reduced to a non-detectable level. One sample (NMRI\_1stInf\_0dpi\_rep1) was excluded because the uninfected control showed unexpected mapping of reads to the \textit{E. falciformis} genome. We consider the three excluded samples to display an uncertain state of infection.

\subsection*{The mouse transcriptome changes upon \emph{E. falciformis} infection}
We here show that upon infection with \textit{E. falciformis}, which induces weight loss and intestinal pathology in mice, the host transcriptome undergoes drastic changes with more than 3000 genes changing their mRNA profile.  Statistical testing for differential expression between infected and uninfected mice revealed changes in mRNA abundance becoming more pronounced (containing more genes) at later time-points post infection (Table 2). 325 mouse mRNAs were considered differently abundant (DA; FDR < 0.01) between controls and 3 dpi, 1,804 mRNAs between controls and 5 dpi and 2,711 mRNAs between controls and 7 dpi. This lead to a combined set of 3,453 unique genes responding to infection (Figure 2bi). DA mRNAs early in infection (3 dpi and 5 dpi) were not a strict subset of genes DA later in infection (7 dpi), which would be the case if the same genes are regulated throughout infection. Instead, the transcriptional profile of the mouse changes more fundamentally with other genes regulated late compared to early in infection. Our results are in agreement with previously published microarray data from the same species. Differences between our controls and 7 dpi samples were correlated with fold-change data obtained from \textit{E. falciformis} infected mice at 6 dpi on Agilent microarrays (\cite{schmidt12}). The data-sets show a strong correlation (Spearman's $\rho$ = 0.74; Figure 2a). Considering both biological differences in the experiments such as exact time-points for sampling, and technical differences between the two methods, this comparison confirms the adequacy of using dual RNA-seq for assessing the host transcriptome.

\subsection*{Epithelial responses vary with mouse immune status}
A distinct response in late infection is indicated by over 1600 DA mRNAs on 7 dpi (Figure 2b). To analyze this result further, we performed hierarchical clustering on the (union of) mouse genes DA between different time-points post infection (Figure 2b and 2c). Three main sample clusters formed (dendrogram of columns at top of Figure 2c). Immune deficient Rag1-/- mice, including infected samples, cluster with control samples. The failure to distinguish between infected and non-infected Rag1-/- samples, confirms the immune deficiency phenotype in these mice and suggest a strong influence of adaptive immune responses on sample and gene clustering. 
We identify a group of genes (cluster 4) which changes transcriptional profile upon infection in immune competent mice only, from high to low abundance. In immune compromised Rag1-/-, another group of genes (cluster 3) display the "infection profile" of immune competent mice also in control samples. These appear to be mRNAs which are kept at a low abundance in healthy animals in a T and B cell dependent manner. Hence, genes in cluster 3 surprisingly appear to depend on functional T and B cells also in uninfected mice (cluster 3) and at the earliest time-point post infection (3 dpi, cluster 4). 
Gene Ontology, GO, terms enriched in cluster 3 are, e.g., "lipid metabolic process" and "protein intraciliary transport". Other enriched terms are regulation of other metabolic processes and "blood coagulation" as well as terms containing "spinal cord", "axon" or "neuronal" regulation or development. (SI file x). We suggest that these are processes which are all intrinsically different in Rag1-/- mice compared to immune competent mice. The genes in cluster 4, which change only in immune competent mice, are highly abundant in controls including Rag1-/- and appear to be down-regulated upon infection by processes which depend on (mature) T and/or B cell activity, also as early as 3 dpi. Enriched GO biological processes include "signalling" and "cell communication" as well as terms containing "external stimuli" and "mating" and as described below, it appears that epithelial healing is induced . 

\subsection*{Epithelium renewal is different depending on immune status}
Several terms enriched in cluster 4 are associated with wound healing and proliferation. 13 terms for cytokines as well as "negative regulation of viral (or inflammatory) response", "negative chemotaxis", "autophagy", "blood coagulation", "inositol phosphate-mediated signaling", and "positive regulation of calcineurin-NFAT" are enriched. mRNAS supporting these terms are less abundant in infected samples. Although speculative, several of these processes can be linked. Inositol signaling can lead to release of calcium and calcineurin-dependent translocation of NFAT to the nucleus and activation of its target genes in T cells, but also many other cell types (reviewed by macian05). GO enrichment also highlights regulation of transforming growth factor$\beta$, TGF$\beta$, epidermal growth factor, EGF, and tumor necrosis factor, TNF. TGF$\beta$ is important for wound healing in intestinal epithelium (beck03), and EGF regulates proliferation of epithelial cells and inhibits apoptosis (suzuki10). TNF is dose-dependent and can suppress inflammatory responses (noti10) and is reported to regulate proliferation of epithelial cells (kaiser97). Additionally, IL-1 and IL-6 regulation are among the enriched GO terms in cluster 4. The IL-1 receptor (type I) is similar to Toll-like receptors and IL-1 induces innate immune responses in many cell types, and influences lymphocyte activity (dinarello09). Both production and secretion appear to be downregulated by the genes is cluster 4. IL-6 has been shown to support repair and inhibit apoptosis after epithelial wounding (kuhn14, probably through the Janus kinase, JAK, and signal transducer and activator of transcription STAT3 (pickert09). IL-6 is also known to be important for development of Th17 responses (ref in kuhn intro) which play an important role in responses to \textit{E. falciformis} (stange--). This analysis suggests that TGF$\beta$, TNF, EGF, IL-1 and IL-6 are main actors in the epithelial response to \textit{E. falciformis} infection and that the response is T and B cell dependent, also at early time-points. Enrichment tests suggest that hosts invest resources in intestinal healing and possible regulatory functions (IL-1 down) also at early time-points of \textit{E. falciformis} infection. The difference here between immunocompetent and immunodeficient Rag1-/- mice suggest that functional T and B cells are needed for these responses. An alternative interpretation is that pathology is lower in Rag1-/- mice and that these responses are therefore not triggered in them. However, based on unpublished data from colleagues (talk to Stange?), severe pathology seen in infected Rag1-/- mice makes this scenario unlikely.

\subsection*{Late infection is strongly enriched for adaptive immunity} 
The pronounced changes late in infection (7 dpi) reflect the expected onset of an adaptive immune response, undelined by enriched GO terms in gene clusters 1 (also abundant in early samples) and 5. Terms such as "antigen binding" and "immunoglobulin receptor binding" (molecular function, MF), and "immune system process", "adaptive immune response" and also "innate immune response" (biological process, BP) are highly enriched in cluster 5 and confirm an activated immune system and adaptive immune responses at this time-point. Among the same genes, natural killer cell regulation, JAK-STAT signalling, and IL-1 and interleukin-2, IL-2 production are enriched biological processes. IL-2 is one target of NFAT signalling and as mentioned above. JAK-STAT signalling can be induced by IL-6. It is likely that the enriched early responses in, e.g., NFAT and IL-6 regulation induce distinct mRNA abundace differences later in infection (7 dpi) and it is encouraging that these links are detected by the methods applied. Taken together, the clustering patterns of immunocompetent versus immunocompromised hosts suggest a strong influence of adaptive immunity on hierarchical clustering, and therefore on the major biological processes which characterize the infection. Using GO enrichment analysis, we additionally show that regulation of IL-1, IL-6, TNF, TGF$\beta$ and EGF are regulated in both first and challenge infections and suggest wound healing as a probable function of this regulation, especially at 7 dpi.
 
\subsection*{RNA processing is enriched in challenge infected immune competent mice} 
Three challenge infected samples (3 dpi, 5 dpi and 7 dpi) from immune competent mice show a distinct profile in cluster 6. This cluster is highly enriched for GO terms for RNA processing and splicing, as well as terms for histone and chromatin modification. \ldots

Include or not?
- daunorubicin metabolic process > cancer drug which prevents DNA replication by stabilizing DNA
- doxorubicin metabolic process  > cancer drug

 
%%%% INPUT
"UDP-N-acetylglucosamine-lysosomal-enzyme" (MF)
%%%%% here %%%%%

\subsection*{Transcriptional differences in the parasite life cycle are independent of mouse immune status}
We present data which supports that \textit{E. falciformis} is "transcriptionally blind" to the host immune status, since comparisons between differently immune-competent mice return no significantly different mRNAs. We performed statistical tests to evaluate significant differences (FDR$<$0.01) in mRNA abundance between different parasite life cycle stages, approximated by time post infection (Table 2). Between early time-points, 3 dpi and 5 dpi, 103 mRNAs were different, whereas between 3 dpi and 7 dpi 1399 mRNAs were DA, and between 5 dpi and 7 dpi 2084 mRNAs were DA (Figure 3a). This indicates that the major changes take place between 5 dpi and 7 dpi, and that variation is smaller between 3 dpi and 5 dpi. This motivated us to define 3 dpi and 5 dpi as "early infection" and 7 dpi as "late infection". Early and late infection samples were tested for DA compared to sporozoites and sporulated oocysts, resulting in 1697 and 3919 DA mRNAs, respectively.
%% Check venn vs table2 and change in ms + code
To evaluate this outcome further, we performed hierarchical clustering of (the union of) the DA mRNAs in the comparisons described above and applied GO enrichment analysis of gene clusters (annotations are inferred from orthologs in other \textit{Eimeria} spp. or \textit{T. gondii}). In the parasite transcriptome, we see no difference between infection in immune competent mice, or T and B cell deficient Rag1-/- mice, or between naive and challenge infected mice. This is surprising considering the measured differences in oocyst output in the same comparisons (Figure 1a), and the fact that these differences are visible in the mouse transcriptomes. Major patters in the parasite transcriptome instead seem to be determined by life cycle stages, independent of the host immune status. Distinct clusters of genes define early infection (3 dpi and 5 dpi) in which schizogony (asexual reproduction) takes place, and separately, late infection (7 dpi) in which it is assumed that gametocytes are present. Extracellular samples of sporozoites and oocysts also cluster separately and are defined by distinct gene clusters. 

\subsection*{Early infection transcriptomes reflect parasite expansion}
Our data suggest ATP production as a dominant feature in sporozoites, followed by biosynthetic processes in the schizogony stages that follow upon invasion. GO enrichment of the major sporozoite defining cluster with high mRNA abundance in this stage (cluster 4) returns different biosynthesis processes. Metabolic pathway analysis (ToxoDB) of the same genes reveals several pathways which suggest that ATP production takes place, including fatty acid degradation, oxidative phosphorylation and  valine, leucine and isoleucine degradation (SI xx). The latter degradation of branched chain amino acids pathway is enriched by genes which have orthologs in \textit{T. gondii}, where they are used to utilize amino acids for energy supply (oppenheim14). In addition, the invasive stage is characterized by "maintenance of protein location in cell" and similar terms. Possibly, this is due to control of microneme protein localization as sporozoites prepare for invasion. In the following life cycle stages, measured at 3 dpi and 5 dpi, in which several rounds of schizogony take place, mRNAs in cluster 6 are abundant in all early samples (except one, discussed below). Among these "early infection" genes in cluster 6, several GO terms (biological process) for biosynthetic acitivity are enriched, e.g., "ribosome biogenesis" and "cellular biosynthetic process", as well as terms for "gene expression" and RNA processing, including terms for tRNAs and ncRNAs. "Cellular amino acid catabolic process" is also enriched. In the small gene cluster 3, no GO terms are enriched. Cluster 6 is distinct in early infection with a biosynthesis profile similar to that of invasive sporozoites. Replication and growth-related processes being enriched highlights the parasite's expansion in numbers on 3 dpi and 5 dpi, as supported both by previous knowledge about the life cycle and our increase in parasite derived sequences (Figure 1b). The mRNAs supporting the processes described here all have a low abundance late in infection, on 7 dpi. This can be explained by a switch from the early asexual expansion towards sexual reproduction on the time-point one day before oocyst outputs peak.

\subsection*{Gametocytes likely determine transcriptome late in infection}
Two gene clusters have a distinct profile with high mRNA abundance on 7 dpi (clusters 2 and 7). Both clusters display low mRNA abundance in other life cycle stages, especially in oocysts and sporozoites. Enriched GO terms such as "movement of cell or subcellular component" and "microtubule-based movement" along with terms suggesting ATP production ("ATP generation from ADP") indicate the presence of motile and energy demanding gametocytes in these samples. Surprisingly, other genes seem to support ATP production in this life-cycle stage compared to sporozoites. Oocyst generation is probably also reflected in terms in cluster 2 and 7: different (peptide/nitrogen compound/cellular protein/macromolecule) "biosynthetic process" terms along with "chitin metabolic process" suggest that the parasite is producing building blocks for oocysts. This would fit the timing of oocyst output which peaks at 8-9 dpi. 
%%%(double check) 
In addition, cluster 2 is enriched for a number of GO terms for "blood coagulation" and reflect the presence of Thrombospondin type I domains in the protein products of cluster 2 mRNAs. Thrombospondin type 1 domains have been reported in \textit{E. tenella} microneme localizing proteins, MIC, e.g. MIC4 (Tomley01). MIC4 mRNA was reported in \textit{E. tenella} sporozoites where it localizes to the apical end, and in late schizonts and late oocyst stages, when sporozoites are forming. This has so far not been demonstrated in \textit{E. falciformis}. Possibly, the mouse parasite prepares for invasion already in gametocytes stages or during oocyst formation, however this is speculative.

\subsection*{Oocysts are characterized by stress responses and differentiation}
Clusters 1 and 5 contain mRNAs with high abundance in oocysts and all other gene clusters have low mRNA abundance in this stage. Cluster 5 is enriched for terms related to stress responses, "DNA repair", "protein modification process" and for "cell differentiation". GO enrichment in cluster 1 contains only one term (adj. p-value 0.11) for "DNA-templated transcription, initiation". Stress responses and DNA repair can be a result of storage in potassium dichromate of mouse faeces with oocysts before purification. Initiation of transcription and cell differentiation probably reflects preparation for invasion when oocysts are taken out of storage and purified for RNA extraction. Overall, the oocyst profile in five out of seven gene clusters is characterized by below average abundance of mRNAs, as can be expected in this inactive life cycle stage which endures long-term survival outside the host. Taken together, sample and gene clustering indicates that genes in clusters 6 are abundant only early in infection and are candidates for merozoite specific genes, whereas clusters 2 and 7 might be useful to characterise gametocytes. Genes in cluster 4 can be considered sporozoite specific, and clusters 1 and 5 as oocyst specific genes in \textit{E. falciformis}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% comment on weird day 7 day 3 clustering
\subsection*{Imperfect clustering might reflect true biological differences}
Hierarchical clustering analysis in most cases does not cluster replicates together, which at a first glance might suggest technical problems. On the other hand, very distinct samples such as oocysts and sporozoites do cluster as replicates, and all 7 dpi samples form one cluster. Also, considering the early/late infection patterns in the parasite data and that these fit well with previous knowledge about asynchronous schizogony and gametocyte formation around 7 dpi, it is worthwhile to consider whether replicate separation reflects true biological variation. It is perceivable that the parasite accommodates to host variation with small differences in, e.g., host stress levels due to litter mates, draught, differences in light exposure or other factors which may vary also in a controlled animal facility. Parasite adjustment to such factors could explain the transcriptional profiles we see, with distinct overall patterns but replicate separation in the asexual phase. We suggest that considering such possibilities is useful for interpreting results and understanding basic biology of the parasite, and, in extension better understand infections in less homogeneous hosts than laboratory mice. 

\subsection*{Evolutionary conservation in life cycle-characteristic gene groups}
A good understanding for similarities and differences between closely related parasites will be useful to draw parallels between the species, and importantly also to avoid incorrect assumptions about, e.g., gene function. One way to achieve such understanding is to identify which genes are species-specific, phylum specific or broadly conserved. To this aim, we tested gene clusters in Figure 3c for enrichment of gene orthologs which are present in other selected species. Groups analyzed for orthologs are: i) \textit{E. falciformis} only ("Efalci"), ii) 10 apicomplexan parasites ("Api") (see additional file xxx), iii) Api, but excluding \textit{Cryprosporidium hominis} ("Api minus C.h."), iv) three \textit{Eimeria} species: \textit{E. falciformis, E. maxima} and \textit{E. tenella} ("Eimeria"), and v) one conserved group containing a broad range of species with, e.g., \textit{Saccharomyces cerevisiae} and \textit{Arabidopsis thaliana} ("conserved"). 
We identify sporozoites as the most \textit{E. falciformis} specific life cycle stage. Sporozoite characteristic genes are enriched for \textit{E. falciformis} unique genes, and the group "conserved" is underrepresented among these genes, further supporting \textit{E. falciformis}-specificity in the sporozoite stage. Gene clusters characteristic for early infection are enriched for "Api" (cluster 6), probably reflecting that similar processes and genes are involved in asexual reproduction of the selected apicomplexan species. This could have implications for, e.g., the use of stage specific markers or drugs which target the asexual stage of the parasite (examples?....). 
Late infection (cluster 2) genes are underrepresented for conserved and "Api minus C.h", indicating that the Eimeria and Efalci categories dominate among these genes. The other characteristic cluster for late infection (cluster 7) is enriched for "conserved" orthologs. This clustering and difference in enrichment of conserved versus apicomplexan genes (minus \textit{C. hominis}) probably reflects simultaneous activity of generic and phylum specific processes in gametes or oocyst formation. 
One oocyst cluster (cluster 1) is underrepresented for "Eimeria". It is possible that this result is driven by high presence of Efalci and Api or Api minus C.h. genes, which are not orthologs with the chicken \textit{Eimeria} species. The other oocyst genes (cluster 5) are enriched for "Api" and are therefore likely oocyst genes which are important for all apicomplexans analyzed here. 

In addition to analysing conservation by ortholog enrichment, we also performed Spearman's correlation analysis between our RNA-seq transcriptomes and RNA-seq data from related parasites. Two datasets for the economically important chicken parasite \textit{E. tenella} (walker15 and reid14) and one dataset of the model apicompomplexan parasite \textit{Toxoplasma gondii} (hehl15) were included in the comparison. Interestingly, this analysis confirms the species specificity for the sporozoite transcriptome, by clustering \textit{E. tenella} sporozoite samples together, but \textit{E. falciformis} sporozoites with \textit{E. falciformis} early infection samples. \textit{E. falciformis} late infection samples correlate the most with \textit{E. tenella} gametocytes, indicating similarity also between the species in this stage and supporting the presence of gametocytes in our samples. \textit{E. tenella} merozoites from both independent studies are most similar to early \textit{E. falciformis} samples, indicating similarity also during asexual reproduction which is also shown by the conservation analysis (Figure?/Table?). We speculate that this clustering is driven by genes in cluster 6 (Figure 3c) since they determine early stages and are enriched for shared apicomplexa orthologs. \textit{E. falciformis} oocysts cluster with unsporulated \textit{E. tenella} oocysts, whereas \textit{E. tenella} sporulated oocysts are most similar to \textit{E. tenella} sporozoites. We have identified groups of genes which are shared among apicomplexa and linked them to the asexual reproduction phase of \textit{E. falciformis}. The analysis also returns \textit{E. falciformis} specific genes, which are abundant in sporozoites. One group of \textit{E. falciformis} oocyst genes (cluster 1, Figure 3c) are underrepresented for \textit{E. tenella} orthologs and can perhaps be used to understand differences between mammalian and avian \textit{Eimeria} spp. oocysts.

\subsection*{Conclusions}
In summary, we have performed the first dual RNA-seq transcriptome of the full life-cycle of an apicomplexan parasite in its natural host. Our analysis of differentially abundant mRNAs at different time-points post infection highlights large groups of genes which characterize the different life stages of the parasite. The dual approach adds insight into the host responses to this intracellular infection, e.g., demonstrating that the host transcriptome is different in a first versus challenge infection, whereas the parasite seems to be transcriptionally "unaware" of the differences in mouse immune competence. 
We show that the transcriptional profile of sporozoites is the most \textit{E. falciformis}-specific stage. The sporozoite defining gene group is characterized by ATP production, regulation of protein localization and biosynthetic processes. The analysis further demonstrates that early infection is not synchronized between samples, neither in host nor parasite. For the parasite, the conformity which is seen in sporozoites and oocysts is however regained late in infection, where both parasite and host transcriptomes are synchronized between samples. We speculate that parasites during asexual replication are more asynchronous than developing gametes and oocysts on 7 dpi. The distinct profile on 7 dpi in both host and parasite suggests that synchronization is driven by interactive processes between the species. Whether gamete formation triggers the adaptive immune responses seen in the host, or previously activated immune responses lead to gamete and oocyst formation remains to be investigated.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	METHODS
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Methods}
\subsection*{Mice and infection procedure}
Three strains of mice were used in our experiments: NMRI (Charles
River Laboratories, Sulzfeld, Germany), C57BL/6 (), and Rag1-/- on
C57BL/6 background (gift from Susanne Hartmann, FU?).  Animal
procedures were performed according to the German Animal Protection
Laws as directed and approved by the overseeing authority Landesamt
fuer Gesundheit und Soziales (Berlin, Germany). Animals where infected
as described by Schmid et al., (schmid12), but tapwater was used
instead of PBS for administration of oocysts. Briefly, NMRI mice were
infected two times, which will be referred to as first and second
infection. For the first infection, 150 sporulated oocysts were
administered in 100 $\mu$L by oral gavage. During the first infection of
60 mice, all animals were weighed every day. On day zero, before
infection, as well as on 3 dpi, 5 dpi and 7 dpi, caeca from 3-4 sacrificed mice per time point were
collected. Epithelial cells were isolated as described in Schmid et
al.(schmid12). For challenge infection, mice recovered for four weeks
before second infection.  Recovery was monitored by weighing and
visual inspection of fur. For the second infection, 1500 sporulated
oocysts were applied by oral gavage. Three mice were used as
non-second infection control, referred to as day 0, second infection.

\subsection*{Oocyst purification for infection and sequencing}
Sporulated oocysts were purified by flotation from feces stored in
potassium dichromate and administered orally in 100 $\mu$L tapwater. One
\textit{E. falciformis} isolate, \textit{E. falciformis} Bayer
Haberkorn 1970, was used for all infections and parasite samples. The
strain is maintained through passage in NMRI mice in our facilities as
described elsewhere (schmid12).

\subsection*{Sporozoite isolation}
Sporozoites were isolated from sporocysts by in vitro excystation. For this,
sporocysts were incubated at 37\newcommand{\degree}C in DMEM containing 0.04\%
tauroglycocholate (MP Biomedicals) and 0.25\% trypsin (Applichem) for
30 min. Sporozoites were purified by the method of Schmatz et al
(schmatz--).

\subsection*{RNA extraction}
Total RNA was isolated from infected epithelial cells, sporozoites and
sporulated oocysts using Trizol according to the manufacturer’s
protocol (Invitrogen). High quality \emph{what is the meaning of 'high
  quality' here?}  RNA was used to produce an mRNA library using the
Illumina’s TruSeq RNA Sample Preparation guide. Sporozoites were stored in 1 mL Trizol until
RNA-isolation. Total RNA was isolated using the PureLink RNA Mini Kit
(Invitrogen) and reverse transcribed into cDNA.

\subsection*{Sequencing, sequence quality assessment and alignment}
cDNA samples were sequenced by either GAIIX or Illumina Hiseq 2000 as
specified in SI xx (both unstranded). A fastq\_quality\_filter
(FASTQ-toolkit, version 0.0.14, available at
https://github.com/agordon/fastx\_toolkit.git) was applied to Illumina
Hiseq 2000 samples after replacing "N" s by "." annotation.  A
phred score of 10 was applied. We further set q = 60. These settings
require that nine out of ten bases or more are correct in at least
60\% of the bases for each read.

\subsection*{Alignment and reference genomes}
We used the published \textit{Mus musculus} mm10 assembly (Genome
Reference Consortium Mouse Build 38, GCA\_000001635.2) as reference
genome including annotations for mouse data. The
\textit{E. falciformis} genome (Heitlinger14) was downloaded from
ToxoDB (Gajria07). For the alignment, the mouse and parasite genome
files were merged into a dual reference genome, and files including
mRNA sequences from both species were aligned against the dual
reference genome using 

###adjust### 
TopHat2 (version 2.0.14, Trapnell09) with -G
specified, and a ####Bowtie2##### (version 1.1.2, Langmead12) index of the dual
genome. Single-end and pair-end sequence samples were aligned
separately with library type 'fr-unstranded' specified for pair-end
samples. Import into R was enabled by the R package Ballgown, which
requires bam files to be processed by Tablemaker (Frazee15), in our
case used with -qW -G specified. Tablemaker in turn makes use of
Cufflinks (version 2.1.1, Trapnell10).
###########################################

\subsection*{Differential mRNA abundance, data normalisation and sample exclusions}
Count data was normalized using the R-package edgeR (version 3.14.0;
cite) with the upperquartile normalisation method. Briefly, genes with
zero coverage in all samples (libraries) are removed and normalisation
factors are calculated for the 75\% quantile for each library. This
normalisation is suitable for read densities following a negative
binomial distribution. Two samples contradicted this assumption
(parasite data) for later modelling and both mouse and parasite data
from these samples were excluded from further analysis:
NMRI\_1st\_3dpi\_rep1 and NMRI\_2nd\_5dpi\_rep1 (SI ...). The method
then fits a generalized linear model (GLM with a negative binomial
link function) for each gene (glmFit) and then performs likelihood
ratio tests for models w or w/o focal factor (glmLRT).

\subsection*{Selection of differentially abundant mRNAs and hierarchical clustering}
A selection of differently abundant mRNAs are used for hierarchical
clustering of \textit{E. falciformis} life cycle relevant genes. In
each comparison (see Table 3), the union of significantly different genes (mRNAs)
were selected. In the next step, the mRNAs from each comparison are
joined. In heatmaps, all samples, i.e., also samples which did not have any
significantly different mRNAs according to our selection, were
included in hierarchical clustering. Scale bar in heatmaps shows 0 as
mean mRNA abundance for each gene (row). Up- (green) and
down-regulation (brown) denote number of standard deviations from 0,
i.e., row mean. Hierarchical clustering was performed using
Euclidean distances, by the complete linkage method ('complete', R package
base).

All (### alignments too?) ### analyses were performed in R (cite R-core). Complete scripts are
available at https://github.com/derele/Ef\_RNAseq.git tagged as version
1.0.
%% We will make this available as a github webpage!!!



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
Animal and parasite experiments: Simone Spork, experimental design: Richard Lucius, Simone Spork, Emanuel Heitlinger, RNA sequecing: Christoph Dieterich, data anylsis: Emanuel Heitlinger, Totta Kasemo, text: all?

\section*{Acknowledgements}
This project was funded by..... Totta Kasemo's PhD project is funded by the German Research Foundation (DFG) program GRK4026: \textit{Parasite Infections: From Experimental Models to Natural Systems}. 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{spbasic} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{ef-rnaseq-refs.bib} %{bmc_article}      % Bibliography file (usually '*.bib' )
% for author-year bibliography (bmc-mathphys or spbasic)
% a) write to bib file (bmc-mathphys only)
% @settings{label, options="nameyear"}
% b) uncomment next line
%\nocite{label}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\setlength{\tabcolsep}{10pt}
\begin{table}[H]
\small
\begin{center}
\caption{Genes used for hierarchical clustering of mRNAs differently abundant depending on time p.i..}
\begin{tabular}{*3l}    \toprule
	\textit{Data description} & \emph{E. falciformis} genes & Mouse genes	\\ \midrule
	Sum of 1st infection NMRI sample differences	& 4 & 8052 \\ 
	(including oocysts and sporozoites if appl.)	\\	
	Used in hierarchical clustering (heatmap)  	& 1618 & 1313 \\ 	\bottomrule	
\hline
\end{tabular}
\end{center}
%\end{adjustwidth}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\newcommand{\bcell}[2][c]{%
  \begin{tabular}[#1]{@{}c@{}}#2\end{tabular}}

\definecolor{LightCyan}{rgb}{0.88,1,1}
\definecolor{LightRed}{rgb}{1,0.88,1}

% Fri Aug 12 11:05:43 2016
\begin{table}[ht]
\centering
\hspace*{-2.5cm}\begin{tabular}{lllllllll}
  \hline
Sample* & \bcell{Sequencing\\method} & batch & \bcell{total\\reads} & \bcell{reads\\mapping\\Mouse} & \bcell{reads\\mapping\\\textit{E. falciformis}} & \bcell{Percentage\\\textit{E. falciformis}}** & \bcell{detected\\ \textit{E. falciformis}\\genes} \\ 
  \hline
NMRI\_2ndInf\_0dpi\_rep1 & GAII & 2 & 108,937,797 & 70,489,674 & 247 & 0.0004 & 1 \\ 
  Rag\_1stInf\_0dpi\_rep1 & hiseq & 3 & 25,362,793 & 18,853,850 & 443 & 0.0023 & 2 \\ 
  C57BL6\_1stInf\_0dpi\_rep1 & hiseq & 3 & 35,731,249 & 25,119,348 & 457 & 0.0018 & 2 \\ 
  C57BL6\_1stInf\_0dpi\_rep2 & hiseq & 3 & 47,085,959 & 34,377,133 & 608 & 0.0018 & 2 \\ 
  Rag\_1stInf\_0dpi\_rep2 & hiseq & 3 & 46,556,156 & 35,233,327 & 676 & 0.0019 & 2 \\ 
  NMRI\_2ndInf\_0dpi\_rep2 & hiseq & 3 & 58,122,244 & 40,794,245 & 3,406 & 0.0083 & 51 \\ 
  \rowcolor{LightCyan}
  NMRI\_2ndInf\_3dpi\_rep1 & hiseq & 3 & 57,934,016 & 40,544,287 & 4,803 & 0.0118 & 95 \\ 
  \rowcolor{LightCyan}
  NMRI\_2ndInf\_5dpi\_rep2 & hiseq & x & 63,965,539 & 48,289,181 & 10,941 & 0.0227 & 407 \\ 
  \rowcolor{LightRed}
  NMRI\_1stInf\_0dpi\_rep1 & GAII & 1 & 82,364,585 & 55,176,243 & 17,954 & 0.0325 & 701 \\ 
  NMRI\_2ndInf\_3dpi\_rep2 & hiseq & 3 & 65,548,826 & 46,171,909 & 29,548 & 0.0640 & 1,580 \\ 
  NMRI\_2ndInf\_7dpi\_rep2 & hiseq & 3 & 67,487,466 & 51,722,265 & 40,091 & 0.0775 & 1,836 \\ 
  Rag\_1stInf\_5dpi\_rep1 & hiseq & 3 & 38,651,359 & 29,982,453 & 63,024 & 0.2098 & 2,548 \\ 
  Rag\_1stInf\_5dpi\_rep2 & hiseq & 3 & 34,779,832 & 25,297,803 & 99,000 & 0.3898 & 2,828 \\ 
  C57BL6\_1stInf\_5dpi\_rep1 & hiseq & 3 & 40,904,388 & 29,319,604 & 185,969 & 0.6303 & 4,173 \\ 
  Rag\_2ndInf\_5dpi\_rep1 & hiseq & 3 & 50,049,848 & 37,093,621 & 192,856 & 0.5172 & 4,167 \\ 
  C57BL6\_1stInf\_5dpi\_rep2 & hiseq & 3 & 29,511,368 & 18,062,349 & 215,696 & 1.1801 & 3,823 \\ 
  C57BL6\_2ndInf\_5dpi\_rep1 & hiseq & 3 & 35,148,432 & 25,660,184 & 262,909 & 1.0142 & 4,563 \\ 
  NMRI\_1stInf\_3dpi\_rep1 & GAII & 1 & 73,236,430 & 49,993,358 & 394,384 & 0.7827 & 5,220 \\ 
  NMRI\_1stInf\_3dpi\_rep2 & GAII & 2 & 160,709,694 & 117,791,044 & 413,051 & 0.3494 & 4,862 \\ 
  NMRI\_1stInf\_5dpi\_rep2 & GAII & 2 & 119,902,722 & 76,419,774 & 794,570 & 1.0290 & 5,333 \\ 
  NMRI\_2ndInf\_5dpi\_rep1 & GAII & 2 & 230,773,955 & 143,186,486 & 1,846,840 & 1.2734 & 5,533 \\ 
  NMRI\_2ndInf\_7dpi\_rep1 & hiseq & 3 & 70,366,762 & 41,467,146 & 8,634,201 & 17.2335 & 5,875 \\ 
  NMRI\_1stInf\_5dpi\_rep1 & GAII & 2 & 76,702,168 & 47,037,087 & 8,669,701 & 15.5631 & 5,700 \\ 
  NMRI\_sporozoites\_rep2 & GAII & 0 & 19,551,681 & 8,656 & 11,470,604 & 99.9246 & 5,513 \\ 
  NMRI\_1stInf\_5dpi\_rep3 & GAII & 0 & 191,099,180 & 83,735,624 & 27,839,458 & 24.9513 & 5,784 \\ 
  NMRI\_1stInf\_7dpi\_rep1 & GAII & 1 & 66,505,514 & 3,310,666 & 39,400,884 & 92.2488 & 5,932 \\ 
  NMRI\_sporozoites\_rep1 & GAII & 1 & 67,325,397 & 4,334 & 43,774,401 & 99.9901 & 5,825 \\ 
  NMRI\_oocysts\_rep1 & GAII & 1 & 68,859,802 & 3,805 & 49,653,065 & 99.9923 & 5,695 \\ 
  NMRI\_oocysts\_rep2 & GAII & 0 & 151,090,783 & 18,524 & 71,019,860 & 99.9739 & 5,777 \\ 
  NMRI\_1stInf\_7dpi\_rep2 & GAII & 1 & 139,749,046 & 21,699,324 & 73,539,445 & 77.2159 & 5,943 \\ 
   \hline
\end{tabular}
\begin{tablenotes}[flushleft]\footnotesize\singlespacing
\item{*} sample names are given as a$)$ mouse strain b$)$ first or challenge
  infection c$)$ days post infection (dpi) and d$)$ replicate number
  seperated by undersocre . \\
\item{**} percentag mapping \textit{E. falciformis} is given as percentage in total mapping reads
\end{tablenotes}
\end{table}
\hspace*{+2.5cm}





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% 	NUMBER OF GENES DIFFERENT IN DIFFERENT COMPARISONS	
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% I like this table. Looks good and transprorts the right kind of
%% information

\clearpage
\section*{mRNA abundance differences between different experimental groups}
\setlength{\tabcolsep}{8pt}
\begin{table}[H]
\small
\begin{center}
\caption{mRNA abundance differences between different experimental groups.}
%\label{tab:table3}
\begin{tabular}{*3l}    \toprule
\textit{Day post infection} & \textit{Ef} genes different & Mouse genes different \\ 
	\textit{comparisons} 	    & (FDR$\leq$1\%) &  (FDR$\leq$1\%/5\%) \\ \midrule
	NMRI 0 vs NMRI 3		& NA   & 274 \\
	NMRI 0 vs NMRI 5		& NA   & 1736 \\
	NMRI 0 vs NMRI 7		& NA   & 2802 \\
	NMRI 3 vs NMRI 5     		& 111  & 1 \\
	NMRI 3 vs NMRI 7  		& 1385 & 1407 \\ 
	NMRI 5 vs NMRI 7  		& 1895 & 873 \\ 
	C57BL/6 0 vs C57BL/6 5		& NA	& 914 \\
	Rag1-/vs Rag1-/- 5		& NA	& 45 \\ \midrule
\textit{Day post infection,} & 		 & 		 \\ 
\textit{parasite relevant comparisons} 	    & 		&  \\ \midrule
	Oocysts vs NMRI 3  	& 3310 & NA \\  
	Oocysts vs NMRI 5	& 3605 & NA \\ 
	Oocysts vs NMRI 7	& 3085 & NA \\ 
	Oocysts vs sporozoites  & 3421 & NA \\
	Sporozoites vs NMRI 3 	& 1663 & NA \\
	Sporozoites vs NMRI 5 	& 1605 & NA \\
	Sporozoites vs NMRI 7 	& 2473 & NA \\ \midrule
\textit{First and second infection} & 		 & 	 \\ 
\textit{comparisons} 	    & 		& 	\\ \midrule
	NMRI 3 1st vs NMRI 3 2nd  	& 0  & 5 \\
	NMRI 5 1st vs NMRI 5 2nd  	& 0  & 1 \\
	NMRI 7 1st vs NMRI 7 2nd  	& 0  & 902 \\
	C57BL/6 1st vs C57BL/6 2nd (day 5) & 0 &  mouse \\
	Rag1-/- 1st vs Rag1-/- 2nd (day 5) & 0 & mouse  \\ \midrule
\bottomrule
	\hline
\end{tabular}
\end{center}
\end{table}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Raw and normalized counts }
  Raw counts of reads mappins to the \textit{E. falciformis} and mouse
  genome for individual samples in our study. Normalized counts for
  seperately for the host and parasite mappings (three compressed csv
  files).
  
  \subsection*{Additional file 2 --- Results of statistical tests (edgeR) }
  Focal contrast, fold-changes, likelihood ratio in/excluding this
  difference in models, p-values , and false discovery rates (adjusted
  p-values) are given for all tested contrasts (one compressed csv
  file).
  
  \subsection*{Additional file 3 --- Additional methods and resuts }
  % this would be cool but requires special referencing in the text
  % (hard to keep track and hard to fit in the journals referencing
  % system). But we can try.
  Document containing additonal figures and summary tables (pdf).

\subsection*{Additional file 4 --- Results of enrichment analyses (topGO)}
  Tables listing all tested gene sets and resulting significatn GO
  terms.
  
  
\end{backmatter}
\end{document}